extension/pom.xml

```diff
@@ -58,7 +63,14 @@
 	        <artifactId>h2</artifactId>
 			<scope>test</scope>
 		</dependency>
-		
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+			<scope>provided</scope>
+			<optional>true</optional>
+		</dependency>
+
 	</dependencies>
```

Comment:
Sure. I'll remove the dependency.

But just to be clear, lombok is a 'provided' dependency so it won't even be included in the final jar (not even in the shaded jar). The generation of equals/hashcode happens at build time by annotation processors so at the bytecode level, the final class will look the same with or without lombok.

But since this is a class that we have complete control over, I guess it should be ok to just add a note mentioning that when adding new fields, we should update equals/hashcode as well. Will update the PR (Will also rename the classes as suggested in the other comment).